python/lsst/ts/FiberSpectrograph/dataManager.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of ts_FiberSpectrograph.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SpectrographData", "DataManager"]
+
+import dataclasses
+import os.path
+
+import astropy.time
+import astropy.io.fits
+import numpy as np
+
+
+@dataclasses.dataclass
+class SpectrographData:
+    """Class to hold data and metadata from a fiber spectrograph.
+    """
+    wavelength: np.ndarray
+    """The wavelength array produced by the instrument in nm."""
+    spectrum: np.ndarray
+    """The flux array in instrumental units."""
+    duration: float
+    """The duration of the exposure in seconds."""
+    date_begin: astropy.time
+    """The start time of the exposure (TAI)."""
+    date_end: astropy.time
+    """The end time of the exposure (TAI)."""
+    type: str
+    """The measurement type (see the XML `expose.type` value)."""
+    source: str
+    """The light source that was measured (see the XML `expose.source` field).
+    """
+    temperature: float
+    """The internal spectrograph temperature."""
```

Comment:
Although that is a good idea, it doesn't much us when we write a FITS file, since it doesn't look like there is a way to write the quantity objects to a header (instead of a binary table). I've filed an astropy issue about this:

https://github.com/astropy/astropy/issues/9332

Reply:
In the short term you could put the unit in a comment yourself -- comments are not currently being used and your code to build the fits header could check for quantities and behave properly.

Reply:
I'd rather not add custom unit handling here: we might as well let astropy handle it directly, if/when they implement support for the above issue.